torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -318,9 +356,17 @@ def identify_mutated_tensors(kernel, kwargs):
         if not config.optimize_user_defined_triton_kernels:
             raise Exception("optimize_user_defined_triton_kernels is False")
 
-        ttir, tensor_param_locs = generate_ttir(kernel, kwargs)
-        ops = parse_ttir(ttir, kwargs)
-        return analyze_kernel_mutations(ops, kwargs, tensor_param_locs)
+        ttir, ordered_tensor_names = generate_ttir(kernel, kwargs)
+        functions = parse_ttir(ttir, kwargs)
+
+        kernel_name = next(iter(functions.keys()))
```

Comment:
Are we sure that the functions in TTIR are always in a topological order?

Reply:
The kernel that is invoked is always the first and dicts are sorted by insertion order. This is not topo sorted, we just know the first element.